### PR TITLE
Add support for styled RSS feeds

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Adds a new `stylesheet` option to the `rss()` function
- Automatically injects [Pretty Feed v3](https://github.com/genmon/aboutfeeds/blob/main/tools/pretty-feed-v3.xsl) if `true`.
- References `public/` file if a `string` value is used

## Testing

Looks like we don't have RSS tests, would like to add

## Docs

Added new section to RSS doc!

### CODE REVIEW

**Comment by matthewp** [comment]
👍

File: docs/src/pages/en/guides/rss.md [hunk 1]
```diff
@@ -41,3 +43,11 @@ export async function getStaticPaths({rss}) {
 ```
 
 Note: RSS feeds will **not** be built during development. Currently, RSS feeds are only generated during your final build.
+
+### Styling
+
+RSS Feeds can be styled with an XSL stylesheet for a more pleasant user experience when they are opened directly in a browser. By default, Astro does not set a stylesheet for RSS feeds, but it can be enabled by setting the `stylesheet` option.
```
Comment (jasikpark) [inline docs/src/pages/en/guides/rss.md:49]: Mind adding a link to https://developer.mozilla.org/en-US/docs/Web/XSLT / something like https://natclark.com/tutorials/xslt-style-rss-feed/ ?

**Review by jasikpark — APPROVED** [review]
wooooo fancy rss feeds :D @genmon might be interested in this

**Comment by jasikpark** [comment]
https://github.com/genmon/aboutfeeds if there's anything to tweak about the XSLT 🤔

**Comment by jasikpark** [comment]
https://natclark.com/index.xml is definitely a nice example - it makes me want to make my own custom XSLT file 👀

**Comment by matthewp** [comment]
Arg, the test failure is one of mine. It seems to be flakey on windows. You can .skip to unblock yourself if you want.

**Review by jonathantneal — APPROVED** [review]
I recommend adding @jasikpark’s suggestion in a follow. Since this is good, we can add these as-is now.

**Comment by jonathantneal** [comment]
Running the tests again. I hope it was a fluke.

**Comment by jonathantneal** [comment]
Tests passed. Here we go!

**Pull request merged**